jsonnet/thanos-ruler.libsonnet

```diff
@@ -1,457 +1,458 @@
           scheme: 'https',
-          api_version: 'v2',
-        }],
-      }),
-    },
-  },
-
-  // query config which holds http configuration
-  // for communication between thanos ruler and thanos querier.
-  queryConfigSecret: {
-    apiVersion: 'v1',
-    kind: 'Secret',
-    metadata: {
-      name: 'thanos-ruler-query-config',
-      namespace: cfg.namespace,
-      labels: {
-        'app.kubernetes.io/name': 'thanos-ruler',
+        }]),
       },
     },
-    type: 'Opaque',
-    data: {},
-    stringData: {
-      'query.yaml': std.manifestYamlDoc([{
-        http_config: {
-          bearer_token_file: '/var/run/secrets/kubernetes.io/serviceaccount/token',
-          tls_config: {
-            ca_file: '/etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt',
-            server_name: 'thanos-querier.openshift-monitoring.svc',
-          },
-        },
-        static_configs: ['thanos-querier.openshift-monitoring.svc:9091'],
-        scheme: 'https',
-      }]),
-    },
-  },
 
-  serviceAccount: {
-    apiVersion: 'v1',
-    kind: 'ServiceAccount',
-    metadata: {
-      name: 'thanos-ruler',
-      namespace: cfg.namespace,
-      annotations: {
-        'serviceaccounts.openshift.io/oauth-redirectreference.thanos-ruler': '{"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"thanos-ruler"}}',
+    serviceAccount: {
+      apiVersion: 'v1',
+      kind: 'ServiceAccount',
+      metadata: {
+        name: 'thanos-ruler',
+        namespace: cfg.namespace,
+        annotations: {
+          'serviceaccounts.openshift.io/oauth-redirectreference.thanos-ruler': '{"kind":"OAuthRedirectReference","apiVersion":"v1","reference":{"kind":"Route","name":"thanos-ruler"}}',
+        },
       },
     },
-  },
 
-  service: {
-    apiVersion: 'v1',
-    kind: 'Service',
-    metadata: {
-      name: 'thanos-ruler',
-      namespace: cfg.namespace,
-      annotations: {
-        'service.beta.openshift.io/serving-cert-secret-name': 'thanos-ruler-tls',
+    service+: {
+      metadata+: {
+        name: 'thanos-ruler',
+        annotations: {
+          'service.beta.openshift.io/serving-cert-secret-name': 'thanos-ruler-tls',
+        },
+        labels: cfg.labels,
       },
-      labels: cfg.labels,
-    },
-    spec: {
-      ports: [{
-        name: 'web',
-        port: cfg.ports.web,
-        targetPort: 'web',
-      }, {
-        name: 'grpc',
-        port: cfg.ports.grpc,
-        targetPort: 'grpc',
-      }],
-      selector: cfg.selectorLabels,
-      sessionAffinity: 'ClientIP',
-      type: 'ClusterIP',
-    },
-  },
-
-  serviceMonitor: {
-    apiVersion: 'monitoring.coreos.com/v1',
-    kind: 'ServiceMonitor',
-    metadata: {
-      name: 'thanos-ruler',
-      namespace: cfg.namespace,
-      labels: {
-        'app.kubernetes.io/name': 'thanos-ruler',
+      spec+: {
+        selector: cfg.selectorLabels,
```

Comment:
we can inherit this from the kube-thanos lib: https://github.com/thanos-io/kube-thanos/blob/main/jsonnet/kube-thanos/kube-thanos-rule.libsonnet#L85